File: src/arduinobot_test/scripts/simple_action_client.py
#!/usr/bin/env pyhthon3
import rospy
import actionlib
from arduinobot_test.msg import FibonacciAction, FibonacciGoal

def fibonacci_client():
    # Create a SimpleAction client
    client = actionlib.SimpleActionClient("fibonacci", FibonacciAction)
    
    # Waits until the action server has started up and began listening
    client.wait_for_server()

    # Creates a goal to send to action server
    goal = FibonacciGoal(goal=20)
   
    # send goal 
    print("sending goal order %s" % goal.goal)
    client.send_goal(goal)
    client.wait_for_result()
    
    return client.get_result()


if __name__ == '__main__':
    # Initialize a ROS node called fibonacci_client
    rospy.init_node("fibonacci_client")
    rospy.loginfo("simple action client started")

    # Keep the node active until the action server returns a result
    result = fibonacci_client()

    print("Result: ", result.sequence)

File: src/arduinobot_test/src/simple_action_client.cpp
#include <ros/ros.h>
#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include "arduinobot_test/FibonacciAction.h"


int main(int argc, char **argv)
{
    ros::init(argc, argv, "fibonacci_client");
    actionlib::SimpleActionClient<arduinobot_test::FibonacciAction> ac("fibonacci");
    ROS_INFO("Waiting for action server to start.");
    ac.waitForServer();

    // create goal class
    ROS_INFO("Action server started, sending a goal");
    arduinobot_test::FibonacciGoal goal;
    goal.goal = 20;

    // Send goal and wait for result
    ac.sendGoal(goal);
    ac.waitForResult(ros::Duration(30.0));

    auto result = ac.getResult();
    ROS_INFO("Action finished: ");
    
    for (int i : result->sequence) {
        ROS_INFO("%d", i);
    }

    return 0;
}